test/unit/modules/delegates.js

```diff
@@ -113,12 +113,133 @@ describe('delegates', () => {
 				});
 			});
 
+			it('should return error if encrypted secret uses unsupported hash algorithm', done => {
+				var accountDetails = {
+					publicKey:
+						'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+					// method is set to 1: MD5
+					encryptedSecret:
+						'$1$rounds=1$5d502e8156eeeaf2acf23d801351dda7$f856ab7731c046da5bf7b9f11fdf20fbc1870845633627b82f84151e432d4e4ca5f69a9c40f15bc2a749e28a66c2e11bbdc45049e0a1f53087fbdd542536c19059e294d231f7aaf30a0600a79ef78519$iv=4a500624cba9c3cb3b6e380556e197d6$tag=24a118a1abab32d256a5de013c20539e$version=1.0.0',
+				};
+
+				config.forging.secret = [accountDetails];
+
+				loadDelegates(err => {
+					expect(err).to.equal(
+						`Invalid encryptedSecret for publicKey: ${accountDetails.publicKey}`
```

Comment:
Shouldn't we receive [this error here?](https://github.com/LiskHQ/lisk/pull/1829/files#diff-9d5cd48f1ba4ec6b2770fd44edc90323R341)